src/server/VRxControl.py

```diff
@@ -0,0 +1,339 @@
+#
+# VRx Control
+#
+
+#
+#
+
+import logging
+from eventmanager import Evt
+
+logger = logging.getLogger(__name__)
+
+class VRxControlManager():
+    def __init__(self, RHData, Events, RACE, nodes, Language, **_kwargs):
+        self.RHData = RHData
+        self.Events = Events
+        self.RACE = RACE
+        self.nodes = nodes
+        self.Language = Language
+
+        self.enabled = True
+        self.controllers = {} # collection of VRxControllers
+
+        self.Events.trigger('VRxC_Initialize', {
+            'registerFn': self.registerController
+            })
+
+        self.Events.on(Evt.STARTUP, 'VRx', self.doStartup)
+        self.Events.on(Evt.HEAT_SET, 'VRx', self.doHeatSet)
+        self.Events.on(Evt.RACE_STAGE, 'VRx', self.doRaceStage, {}, 75)
+        self.Events.on(Evt.RACE_START, 'VRx', self.doRaceStart, {}, 75)
+        self.Events.on(Evt.RACE_FINISH, 'VRx', self.doRaceFinish)
+        self.Events.on(Evt.RACE_STOP, 'VRx', self.doRaceStop)
+        self.Events.on(Evt.RACE_LAP_RECORDED, 'VRx', self.doRaceLapRecorded, {}, 200, True)
+        self.Events.on(Evt.LAPS_CLEAR, 'VRx', self.doLapsClear)
+        self.Events.on(Evt.LAP_DELETE, 'VRx', self.doLapDelete)
+        self.Events.on(Evt.FREQUENCY_SET, 'VRx', self.doFrequencySet, {}, 200, True)
+        self.Events.on(Evt.MESSAGE_INTERRUPT, 'VRx', self.doSendPriorityMessage)
+        self.Events.on(Evt.OPTION_SET, 'VRx', self.doOptionSet)
+
+    def registerController(self, controller):
+        if hasattr(controller, 'name'):
+            if controller.name in self.controllers:
+                logger.warning('Overwriting VRx Controller "{0}"'.format(controller.name))
+
+            controller.manager = self
+            self.controllers[controller.name] = controller
+            logger.info('Importing VRx Controller {}'.format(controller.name))
+        else:
+            logger.warning('Invalid controller')
```

Comment:
This check for just a name looks a bit too limited to be useful, and possibly just defers causing an error later. If you want to make the controller conform to a protocol, use something like module abc (declaring an interface), and check for it being implemented via isinstance